### PR TITLE
adds a filter!/2 function

### PR DESCRIPTION
**Pull request by joshleecreates**
so that I don't have to do this horribleness: https://github.com/localvore-today/data_scribe/pull/1/files/9b0562d467ff8fe099dcee03ed6f662ada472132#r109328120

### CODE REVIEW

File: lib/localvore_sdk.ex [hunk 1]
```diff
@@ -11,6 +11,13 @@ defmodule LocalvoreSdk do
     |> get
   end
 
+  def filter!(resource, filters) do
```
Comment (doomspork) [inline lib/localvore_sdk.ex]: We already have `fetch/2`, right?

```elixir
case filter(resource, filter) do
  {:ok, result} -> result
  {:error, reason} -> raise reason
end
```

File: lib/localvore_sdk.ex [hunk 2]
```diff
@@ -22,7 +29,7 @@ defmodule LocalvoreSdk do
   defp build_url(query, resource), do: resource <> "?" <> query
 
   defp domain do
-    Application.get_env(:localvore_sdk, :api_url, @default_api_url)
+    Application.get_env(:localvore_sdk, :api_url, @api_url)
```
Comment (doomspork) [inline lib/localvore_sdk.ex]: This change has brought this function to my attention, is there a reason we're using functions for this vs something like:

```elixir
@default_api_url "https://api.localvoretoday.com"
@default_api_version "2016-10-28"

@domain Application.get_env(:localvore_sdk, :api_url, @default_api_url)
@version Application.get_env(:localvore_sdk, :api_version, @default_api_version)
```

On a separate note, I think the `default` prefix is nice-to-have so it's clear it's a default.

**Review by doomspork — CHANGES_REQUESTED** [review]
@joshleecreates couple of little changes.

File: lib/localvore_sdk.ex [hunk 3]
```diff
@@ -11,6 +14,13 @@ defmodule LocalvoreSdk do
     |> get
   end
 
+  def filter!(resource, filters) do
+    case filter(resource, filter) do
+      {:ok, result} -> result
+      {:error, reason} -> raise reason
```
Comment (dmvt) [inline lib/localvore_sdk.ex]: We should raise a `HTTPoison.Error` for consistency

**Review by dmvt — CHANGES_REQUESTED** [review]
One small change

File: lib/localvore_sdk.ex [hunk 4]
```diff
@@ -21,9 +31,7 @@ defmodule LocalvoreSdk do
 
   defp build_url(query, resource), do: resource <> "?" <> query
 
-  defp domain do
-    Application.get_env(:localvore_sdk, :api_url, @default_api_url)
-  end
+  defp domain, do: @domain
```
Comment (doomspork) [inline lib/localvore_sdk.ex]: Why do you need a function at all for this or `version/0`?

```elixir
defp process_url(url), do: Enum.join([@domain, @version, url], "/")
```

**Review by doomspork — CHANGES_REQUESTED** [review]
➜  localvore_sdk git:(feature/filter_w_bang) mix deps.get && mix compile --force
Running dependency resolution...
All dependencies up to date
Compiling 1 file (.ex)
warning: variable "filter" does not exist and is being expanded to "filter()", please use parentheses to remove the ambiguity or change the variable name
  lib/localvore_sdk.ex:18

warning: undefined module attribute @api_version, please remove access to @api_version or explicitly set it before access
  lib/localvore_sdk.ex:56: LocalvoreSdk.version/0

warning: module attribute @version was set but never used
  lib/localvore_sdk.ex:8

warning: variable "filters" is unused
  lib/localvore_sdk.ex:17


== Compilation error on file lib/localvore_sdk.ex ==
** (CompileError) lib/localvore_sdk.ex:18: undefined function filter/0
    (stdlib) lists.erl:1338: :lists.foreach/2
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6

File: lib/localvore_sdk.ex (hunk 4 above)
Comment (joshleecreates) [reply lib/localvore_sdk.ex]: So other modules can access it

Comment (doomspork) [reply lib/localvore_sdk.ex]: But it's a private function

Comment (joshleecreates) [reply lib/localvore_sdk.ex]: Doh

**Review by doomspork — CHANGES_REQUESTED** [review]
@joshleecreates I still get warnings and things when I compile this code.  Are you running something like `mix compile --force` before you push for review?

==> localvore_sdk
Compiling 1 file (.ex)
warning: undefined module attribute @default_api_url, please remove access to @default_api_url or explicitly set it before access
  lib/localvore_sdk.ex:25: LocalvoreSdk.domain/0

warning: undefined module attribute @current_api_version, please remove access to @current_api_version or explicitly set it before access
  lib/localvore_sdk.ex:49: LocalvoreSdk.version/0

warning: module attribute @api_version was set but never used
  lib/localvore_sdk.ex:5

warning: module attribute @api_url was set but never used
  lib/localvore_sdk.ex:4

warning: function Poison.decode!/1 is undefined (module Poison is not available)
  lib/localvore_sdk.ex:40

warning: function Exque.Utils.atomize_keys/1 is undefined (module Exque.Utils is not available)
  lib/localvore_sdk.ex:41

**Review by dmvt — APPROVED** [review]

**Review by doomspork — APPROVED** [review]

**Pull request merged**